src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -652,14 +655,17 @@ class ServerSecurityHandshakerFactory : public HandshakerFactory {
 //
 
 RefCountedPtr<Handshaker> SecurityHandshakerCreate(
-    tsi_handshaker* handshaker, grpc_security_connector* connector,
-    const ChannelArgs& args) {
+    absl::StatusOr<tsi_handshaker*> handshaker,
+    grpc_security_connector* connector, const ChannelArgs& args) {
   // If no TSI handshaker was created, return a handshaker that always fails.
   // Otherwise, return a real security handshaker.
-  if (handshaker == nullptr) {
-    return MakeRefCounted<FailHandshaker>();
+  if (!handshaker.ok()) {
+    return MakeRefCounted<FailHandshaker>(handshaker.status());
+  } else if (*handshaker == nullptr) {
+    return MakeRefCounted<FailHandshaker>(absl::UnknownError(
```

Comment:
Please add a TODO that once all TSI implementations are updated to pass a `StatusOr<>` instead of null, we should change this to use `absl::InternalError()`.